Remove property named `deprecated` from linting spec

ee7 opened the pull request:
This PR makes the `configlet lint` spec match these docs:
- [building/tracks/config-json.md](https://github.com/exercism/docs/blob/214c88b7e85e/building/tracks/config-json.md)
- [building/tracks/deprecated-exercises.md](https://github.com/exercism/docs/blob/214c88b7e85e/building/tracks/deprecated-exercises.md)

The property named `deprecated` was replaced by the `status` key having
a value of "deprecated", and no track still uses the property named
`deprecated`.


---

As far as I can tell looking through every track-level `config.json` locally, no track still uses this property. Could a reviewer confirm this? Also, these only show results in the old `exercism/v3` repo:
- https://github.com/search?q=org%3Aexercism+%22deprecated%3A+false%22&type=code
- https://github.com/search?q=org%3Aexercism+%22deprecated%3A+true%22&type=code

The [building/tracks/config-json.md](https://github.com/exercism/docs/blob/214c88b7e85ed924fa6ac637f7d5bbd5fbd0c721/building/tracks/config-json.md) file doesn't mention this property:

> ### Concept exercises
> 
> Each concept exercise is an entry in the `exercises.concept` array.
> Exercises are ordered on the website in the same order they are listed in this file, and should match the typical order in which they should be solved.
> The following fields make up a concept exercise:
> 
> - `uuid`: a V4 UUID that uniquely identifies the exercise. The UUID must be unique both within the track as well as across all tracks
> - `slug`: the exercise's slug, which is a lowercased, kebab-case string. The slug must be unique across all concept _and_ practice exercise slugs within the track. Its length must be <= 255. 
> - `name`: the exercise's name. Its length must be <= 255. 
> - `concepts`: an array of concept slugs that are taught by this concept exercise
> - `prerequisites`: an array of concept slugs that must be unlocked before a student can start this exercise
> - `status` (optional): the exercise's status, which is one of `"wip"`, `"beta"` `"active"`, or `"deprecated"`; defaults to `"active"` if not specified
>   - `wip`: A work-in-progress exercise not ready for public consumption. Exercises with this tag will not be shown to students on the UI or be used for unlocking logic. They may appear for maintainers.
>   - `beta`: This signifies active exercises that are new and which we would like feedback on. We show a beta label on the site for these exercise, with a Call To Action of "Please give us feedback."
>   - `active`: The normal state of active exercises
>   - `deprecated`: Exercises that are no longer shown to students who have not started them (not usable at this stage). See [Deprecated Exercises](./deprecated-exercises) for more information.


> ### Practice exercises
> 
> Each concept exercise is an entry in the `exercises.practice` array. The following fields make up a concept exercise:
> 
> - `uuid`: a V4 UUID that uniquely identifies the exercise. The UUID must be unique both within the track as well as across all tracks
> - `slug`: the exercise's slug, which is a lowercased, kebab-case string. The slug must be unique across all concept _and_ practice exercise slugs within the track. Its length must be <= 255. 
> - `name`: the exercise's name. Its length must be <= 255. 
> - `practices`: an array of concept slugs that the exercise is helping students practice
> - `prerequisites`: an array of concept slugs that must be unlocked before a student can start the exercise
> - `difficulty`: a number indicating the difficulty of the exercise. The number must be in the range of 0 (easiest) to 10 (hardest)
> - `status` (optional): the exercise's status, which is either `"wip"`, `"beta"`, `"active"` or `"deprecated"`; defaults to `"active"` if not specified
>   - `wip`: A work-in-progress exercise not ready for public consumption. Exercises with this tag will not be shown to students on the UI or be used for unlocking logic. They may appear for maintainers.
>   - `beta`: This signifies active exercises that are new and which we would like feedback on. We show a beta label on the site for these exercise, with a Call To Action of "Please give us feedback"
>   - `active`: The normal state of active exercises
>   - `deprecated`: Exercises that are no longer shown to students who have not started them (not usable at this stage).


And searching this repo for "deprecated" with this PR:
```
$ git grep --break --heading 'deprecated'
building/config.json
489:    "slug": "tracks/deprecated-exercises",
490:    "path": "building/tracks/deprecated-exercises.md",

building/configlet/lint.md
86:- The `"exercises.concept[].concepts"` value must be a non-empty array of strings if `"exercises.concept[].status"` is not equal to `deprecated`
87:- The `"exercises.concept[].concepts"` value must be an empty array if `"exercises.concept[].status"` is equal to `deprecated`
93:- The `"exercises.concept[].prerequisites"` value must be a non-empty array of strings if `"exercises.concept[].status"` is not equal to `deprecated`, except for exactly one exercise which _is_ allowed to have an empty array as its value
94:- The `"exercises.concept[].prerequisites"` value must be an empty array if `"exercises.concept[].status"` is equal to `deprecated`
102:- The `"exercises.concept[].status"` value must be the string `wip`, `beta`, `active` or `deprecated`
116:- The `"exercises.practice[].practices"` value must be a non-empty array of strings if `"exercises.practice[].status"` is not equal to `deprecated`
117:- The `"exercises.practice[].practices"` value must be an empty array if `"exercises.practice[].status"` is equal to `deprecated`
123:- The `"exercises.practice[].prerequisites"` value must be a non-empty array of strings if `"exercises.practice[].status"` is not equal to `deprecated`
124:- The `"exercises.practice[].prerequisites"` value must be an empty array if `"exercises.practice[].status"` is equal to `deprecated`
131:- The `"exercises.practice[].status"` value must be the string `wip`, `beta`, `active` or `deprecated`

building/tracks/config-json.md
96:- `status` (optional): the exercise's status, which is one of `"wip"`, `"beta"` `"active"`, or `"deprecated"`; defaults to `"active"` if not specified
100:  - `deprecated`: Exercises that are no longer shown to students who have not started them (not usable at this stage). See [Deprecated Exercises](./deprecated-exercises) for more information.
151:- `status` (optional): the exercise's status, which is either `"wip"`, `"beta"`, `"active"` or `"deprecated"`; defaults to `"active"` if not specified
155:  - `deprecated`: Exercises that are no longer shown to students who have not started them (not usable at this stage).

building/tracks/deprecated-exercises.md
5:Instead, we allow exercises to be deprecated.
12:To help with this, we have config rules that accompany a deprecated exercise
17:- **`status`** set to `deprecated`
25:All other fields in deprecated exercises can be safely changed or removed in accordance with the wider config rules.

dev/badges.md
122:- Where did it go?: completed an exercise that has since been deprecated
```

kytrinyx commented:
Thank you!

jwood803 commented:
Hmmmes, it may be the same issue in the `Installing F#` section as well. Thanks for letting us know @sunzenshen! I can hit these or you can, if you'd like. Whoever submits it first, I guess :stuck_out_tongue:

sunzenshen commented:
Since you noticed the discrepancy first, I'll let you finish your submissions.  Seems like I didn't find everything when searching through the docs project. :)

jwood803 commented:
I updated one other `jpg` but I noticed some `png` images weren't showing either. @kytrinyx should they be `jpg` or would the path be off? I'm just referencing them as they are in the folder structure.

kytrinyx commented:
I don't know what they are--the images are in `docs/img` for the relevant tracks.

kytrinyx commented:
Wait, are you saying that they're referenced as png and they are png, but they're not showing? Which language tracks?

jwood803 commented:
In the F# and perhaps the C# ones. Below is where they are.

![image](https://cloud.githubusercontent.com/assets/1578160/11542881/b9fd3586-9907-11e5-91f8-132822bac005.png)

kytrinyx commented:
That's where they used to be. This documentation is being served from the TRACK_REPO/docs/img/ dirs.

kytrinyx commented:
I'll take a look at the F# and C# repos, though, and see what's going on.

jwood803 commented:
Ya, I'm quite a bit confused :confounded:

I thought I remember reading that it changed a while ago, but it looks like it was the same when I pulled down the latest.

kytrinyx commented:
Hah. Found it. Not in the visualstudio directory :) Fixing that now.

kytrinyx commented:
OK, found the F# one as well. jpg vs jpeg.

ee7 committed:
Remove property named `deprecated` from linting spec

This commit makes the `configlet lint` spec match these docs:
- https://github.com/exercism/docs/blob/214c88b7e85e/building/tracks/config-json.md
- https://github.com/exercism/docs/blob/214c88b7e85e/building/tracks/deprecated-exercises.md

The property named `deprecated` was replaced by the `status` key having
a value of "deprecated", and no track still uses the property named
`deprecated`.
